sbol/identified.py

```diff
@@ -189,3 +189,251 @@ def update_uri(self):
                                 "An object with URI " + self.identity +
                                 " is already in the Document",
                                 SBOLErrorCode.SBOL_ERROR_URI_NOT_UNIQUE)
+
+    def copy(self, target_doc = None, target_namespace = None, version = None):
+
+        def parseNamespace(uri):
+            rlimit = uri.rfind('#') + 1
+            if rlimit:
+                return uri[:rlimit]
+            rlimit = uri.rfind('/') + 1
+            if rlimit:
+                return uri[:rlimit]
+            return ''
```

Comment:
what is the advantage of using a nested function here?

Reply:
Moved to top level